corehq/apps/app_manager/static/app_manager/js/menu.js

```diff
@@ -32,19 +32,39 @@ hqDefine("app_manager/js/menu", [
         });
     };
 
+    // Frequently poll for changes to app, for the sake of showing the "Updates available to publish" banner.
+    // Avoid checking if the user is idle, which here is defined by mouse or keyboard activity.
     var initPublishStatus = function () {
+        var frequency = 20000,
+            lastActivity = (new Date()).getTime(),
+            msSinceLastActivity = function () {
+                return (new Date()).getTime() - lastActivity;
+            },
+            updateLastActivity = function () {
+                if (msSinceLastActivity() > frequency) {
+                    // If they're coming back after long inactivity, do an immediate check
+                    _checkPublishStatus(true);
+                }
+                lastActivity = (new Date()).getTime();
+            };
+
+        updateLastActivity();
+        $(document).on('mousemove keydown', _.throttle(updateLastActivity, 5000));
```

Comment:
`mousemove` would make this get called a lot number of times, not sure how that effects JS performance.
Can we instead use a way to do this when the tab is active, https://stackoverflow.com/questions/1760250/how-to-tell-if-browser-tab-is-active?

Reply:
Had trouble getting it working earlier. Updated: f7039a808b8